packages/core/src/components/event-lake.store.ts

```diff
@@ -27,20 +28,39 @@ export interface EventLakeStorageLayer {
 export class EventLakeStore<Event extends IEsEvent> {
   constructor(
     public readonly storageLayer: EventLakeStorageLayer,
-    public readonly serializer: ISerializer<Event>,
+    public readonly serializer: ISerializer<Event, any>,
     public readonly eventBus?: IEventBus,
   ) {}
 
-  async append(lakeId: LakeId, changes: Event[], trx: Transaction) {
+  async append(
+    lakeId: LakeId,
+    changes: Event[],
+    trx: Transaction,
+  ): Promise<ISavedChange<Event>[]> {
     const serialized = await Promise.all(
       changes.map((change) => this.serializer.serialize(change)),
     );
 
-    trx.onCommit(() => {
-      for (const change of changes) this.eventBus?.publish(change);
+    const result = await this.storageLayer.append(lakeId, serialized, trx);
+
+    const saved = (await Promise.all(
+      result.map((r) => this.serializer.deserialize(r)),
+    )) as ISavedChange<Event>[];
+
+    for (const save of saved) {
+      const matching = changes.find((c) => c.id.equals(save.id));
+      if (matching) {
+        (matching as any).revision = save.revision;
```

Comment:
Don't like the fact that this function mutate the arg changes. 
If it's not on purpose, then I think this for loop is useless as you don't use changes in later code (only saved)

Reply:
Yes but in an "append" function you don't expect a mutation of the args